pkg/controllers/services_controller.go

```
@@ -96,6 +96,83 @@ type ServicesController struct {
 	Clientset *kubernetes.Clientset
 	Services  *ServiceMap
 	Proxy     nat.ProxyProcessor
+
+	// NodeName is the node this instance runs on. Datapath rules are only
+	// programmed for backend pods hosted here. Empty disables the check and
+	// restores the previous cluster-wide behavior.
+	NodeName string
+}
+
+// endpointNode returns the node hosting the endpoint's first address.
+func endpointNode(ep *v1.Endpoints) (string, bool) {
```

golangci-lint (2.12.2)

[error] SA1019: v1.Endpoints is deprecated: This API is deprecated in v1.33+. Use discoveryv1.EndpointSlice. 

(staticcheck)